private/buf/buflsp/buflsp.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package buflsp implements a language server for Protobuf.
+//
+// The main entry-point of this package is the New() function, which creates a new LSP server.
+package buflsp
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/buf/bufctl"
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule"
+	"github.com/bufbuild/buf/private/gen/data/datawkt"
+	"github.com/bufbuild/buf/private/pkg/app/appext"
+	"github.com/bufbuild/buf/private/pkg/tracing"
+	"go.lsp.dev/jsonrpc2"
+	"go.lsp.dev/protocol"
+	"go.uber.org/zap"
+)
+
+// ErrNotInit is returned by LSP server methods that are called without first
+// initializing the server.
+var ErrNotInit = fmt.Errorf("the first call to the server must be the %q method", protocol.MethodInitialize)
+
+// New constructs a new LSP server, ready to begin listening.
+func New(
```

Comment:
NewServer

Reply:
I renamed this to Listen() because it actually starts up a connection.